plugins/libplugin-pay.c

```diff
@@ -3559,6 +3563,8 @@ static void presplit_cb(struct presplit_mod_data *d, struct payment *p)
 			 * they'll be used when aggregating the payments
 			 * again. */
 			c->invstring = tal_strdup(c, p->invstring);
+			if (p->description)
+				c->description = tal_strdup(c, p->description);
```

Comment:
This is really bad, as it replicates the description in all child-payments, bloating the database with duplicate values. I thought we did the sane thing, and didn't copy the `invstring` either, as that is constant in the entire tree, but apparently it got changed somewhere along the line.

There is no point in duplicating these values, and we can use the convention of attaching them only to the root payment.

Reply:
Hmm, fixing this is a significant change.  Let me try... (and add a test!).  I copied the invstring, assuming that was doing The Right Thing.

Reply:
Actually, this was all a mess!  This duplicate is only for the pre-splitter, but it turns out we were often returning listpays() without *any* bolt11.

I've fixed that.